network/network_interface.ml

```diff
@@ -25,299 +25,609 @@ let uri () = "file:" ^ !default_path
+      end in
+      let conservative_p = Param.mk ~name:"conservative" T._conservative_opt_t in
+      let config_p = Param.mk ~name:"config" T._config_t in
+      declare
+        "make_config"
+        ["Make bridge configuration"]
+        (debug_info_p @-> conservative_p @-> config_p @-> unit_p @-> returning unit_p err)
+  end
+
+  module PVS_proxy = struct
+    module Server = struct
+      type t = {
+        uuid: string;
+        addresses: Unix.inet_addr list;
+        first_port: int;
+        last_port: int;
+      } [@@deriving rpcty]
+    end
+
+    module Client = struct
+      type t = {
+        uuid: string;
+        mac: string;
+        interface: string;
+        prepopulate: bool;
+      } [@@deriving rpcty]
+    end
+
+    type t = {
+      site_uuid: string;
+      site_name: string;
+      servers: Server.t list;
+      clients: Client.t list;
+      vdi: string;
+    } [@@deriving rpcty]
+
+    (* 	external configure_site : debug_info -> PVS_proxy.t -> unit = "" *)
+    let configure_site =
+      let pvs_p = Param.mk ~description:["proxy"] t in
+      declare
+        "configure_site"
+        ["Configure site"]
+        (debug_info_p @-> pvs_p @-> returning unit_p err)
+    (* 	external remove_site : debug_info -> string -> unit = "" *)
+    let remove_site =
+      let site_p = Param.mk ~description:["site name"] Types.string in
+      declare
+        "remove_site"
+        ["Remove site"]
+        (debug_info_p @-> site_p @-> returning unit_p err)
+  end
 end
 
-type kind = Openvswitch | Bridge
-type bond_mode = Balance_slb | Active_backup | Lacp
-type fail_mode = Standalone | Secure
 
-let string_of_kind = function
-	| Openvswitch -> "openvswitch"
-	| Bridge -> "bridge"
-
-module Bridge = struct
-	external get_all : debug_info -> unit -> bridge list = ""
-	external create : debug_info -> ?vlan:(bridge * int) ->
-		?mac:string -> ?igmp_snooping:bool -> ?other_config:(string * string) list -> name:bridge -> unit -> unit = ""
-	external destroy : debug_info -> ?force:bool -> name:bridge -> unit -> unit = ""
-	external get_kind : debug_info -> unit -> kind = ""
-	external get_all_ports : debug_info -> ?from_cache:bool -> unit -> (port * iface list) list = ""
-	external get_all_bonds : debug_info -> ?from_cache:bool -> unit -> (port * iface list) list = ""
-	external set_persistent : debug_info -> name:bridge -> value:bool -> unit = ""
-	external add_port : debug_info -> ?bond_mac:string -> bridge:bridge -> name:port -> interfaces:iface list ->
-		?bond_properties:(string * string) list -> ?kind:port_kind -> unit -> unit = ""
-	external remove_port : debug_info -> bridge:bridge -> name:port -> unit = ""
-	external get_interfaces : debug_info -> name:bridge -> iface list = ""
-	external get_physical_interfaces : debug_info -> name:bridge -> iface list = ""
-	external make_config : debug_info -> ?conservative:bool -> config:(bridge * bridge_config_t) list-> unit -> unit = ""
-end
 
-exception PVS_proxy_connection_error
```

Comment:
@jonludlam can we have multiple kind or errors `err`, `pvs_err`, ... so that specific calls are bound to return specific errors? Is it worth doing it?